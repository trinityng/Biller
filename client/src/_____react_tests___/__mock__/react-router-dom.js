import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;