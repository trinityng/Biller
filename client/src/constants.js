export const HOST = 'http://localhost:8000';

export const USER_EMAIL = 'USER_EMAIL';
export const UPDATE_PROFILE = 'UPDATE_PROFILE';
// User Token
export const USER_TOKEN = 'USER_TOKEN';
export const SET_USER_INFO = 'SET_USER_INFO';

// Server URI
export const LOGIN_URI = '/auth/login/';
export const REGISTER_URI = '/auth/register/';
export const PLAID_ACCESS_TOKEN_URI = '/plaid/access-token/';
export const PLAID_TRANSACTIONS_URI = '/plaid/transactions/';
export const PLAID_TRANSACTIONS_EACH_URI =
  '/plaid/transactions-of-each-account/';
export const PLAID_CATEGORIES_URI = '/plaid/categories/';
export const PLAID_NET_WORTH_URI = '/plaid/net-worth/';
export const PLAID_MONTHLY_EXPENSES_URI = '/plaid/monthly-total-expenses/';
export const PLAID_BILLS_URI = '/plaid/bills/';

// Auth
export const LOGIN_FAILED = 'LOGIN_FAILED';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESSFULLY';

export const REGISTER_FAILED = 'REGISTER_FAILED';
export const REGISTER_SUCCESS = 'REGISTER_SUCCESSFULLY';

export const LOGOUT = 'LOGOUT';

// Plaid
export const PLAID_PRODUCT = ['transactions'];
export const PLAID_PUlLIC_TOKEN = 'PLAID_PUlLIC_TOKEN';
export const PLAID_PUlLIC_KEY = '66974676d9f0b1bcf30d24f66881e0';
export const PLAID_DEV_ENV = 'development';
export const PLAID_SB_ENV = 'sandbox';

export const PLAID_LOGIN_SUCCESS = 'PLAID_LOGIN_SUCCESSFULLY';
export const PLAID_LOGIN_FAILED = 'PLAID_LOGIN_FAILED';

export const PLAID_GET_TRANSACTIONS_SUCCESS = 'PLAID_GET_TRANSACTIONS_SUCCESS';
export const PLAID_GET_TRANSACTIONS_FAILED = 'PLAID_GET_TRANSACTION_FAILED';

export const PLAID_GET_TRANSACTIONS_EACH_SUCCESS =
  'PLAID_GET_TRANSACTIONS_EACH_SUCCESS';
export const PLAID_GET_TRANSACTIONS_EACH_FAILED =
  'PLAID_GET_TRANSACTIONS_EACH_FAILED';

export const PLAID_GET_CATEGORY_SUCCESS = 'PLAID_GET_CATEGORY_SUCCESS';
export const PLAID_GET_CATEGORY_FAILED = 'PLAID_GET_CATEGORY_FAILED';

export const PLAID_GET_NET_WORTH_SUCCESS = 'PLAID_GET_NET_WORTH_SUCCESS';
export const PLAID_GET_NET_WORTH_FAILED = 'PLAID_GET_NET_WORTH_FAILED';

export const PLAID_GET_MONTHLY_EXPENSE_SUCCESS =
  'PLAID_GET_MONTHLY_EXPENSE_SUCCESS';
export const PLAID_GET_MONTHLY_EXPENSE_FAILED =
  'PLAID_GET_MONTHLY_EXPENSE_FAILED';

export const PLAID_GET_ALL_BILLS_SUCCESS = 'PLAID_GET_ALL_BILLS_SUCCESS';
export const PLAID_GET_ALL_BILLS_FAILED = 'PLAID_GET_ALL_BILLS_FAILED';
