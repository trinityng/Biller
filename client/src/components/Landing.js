import React from 'react';
import './Landing.css';
import { Link } from 'react-router-dom';
import { Button, Layout, Menu, Breadcrumb } from 'antd';

const { Header, Footer, Content } = Layout;

const Landing = () => {
  return (
    <div style={{ height: '75vh' }} className="container valign-wrapper">
      <div className="row">
        <div className="col s12 center-align">
          <h4>
            <b>Landing Page </b>
          </h4>
          <br />
          <div className="col s6">
            <Link to="/login">
              <Button type="primary"> Login </Button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Landing;
