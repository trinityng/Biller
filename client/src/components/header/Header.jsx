import React from "react";
import { Layout, Menu } from "antd";
import "antd/dist/antd.css";
import { Link } from 'react-router-dom';

const { Header } = Layout;

const HeaderPage = () => {
    return(
        <Header style={{ position: "fixed", zIndex: 1, width: "100%" }}>
            <div/>
            <Menu theme="silver" mode="horizontal" style={{ lineHeight: "64px", float: "right" }} >
            <Menu.Item key="1">
                <Link to="/login">Login</Link>
            </Menu.Item>
            <Menu.Item key="2">
                <Link to="/register">Sign Up</Link>
            </Menu.Item>
            </Menu>
        </Header>
    );
}

export default HeaderPage