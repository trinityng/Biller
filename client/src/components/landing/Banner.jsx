import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import QueueAnim from 'rc-queue-anim';
import { Button } from 'antd';
import BannerImage from './BannerImage';
import { USER_TOKEN } from '../../constants';

class Banner extends React.PureComponent {
  static propTypes = {
    className: PropTypes.string,
  };
  static defaultProps = {
    className: 'home-banner',
  };
  render() {
    const { className } = this.props;
    const token = localStorage.getItem(USER_TOKEN);
    return (
      <div className={`home-layout-wrapper ${className}`}>
        <div className="home-layout">
          <QueueAnim
            className={`${className}-content-wrapper`}
            delay={300}
            ease="easeOutQuart"
          >
            <h1 key="h2">Biller</h1>
            <p key="p">
              Your Personal Bill Assistant. <br />
              Join us for free today and never worry about your bill again
            </p>
            <span key="button">
              <Link to="/register">
                <Button
                  type="primary"
                  shape="round"
                  onClick={() => {
                    window.location.href = '/register';
                  }}
                >
                  Sign up
                </Button>
              </Link>
              {token ? (
                <Link to="/login">
                  <Button
                    type="primary"
                    shape="round"
                    onClick={() => {
                      window.location.href = '/dashboard';
                    }}
                  >
                    Dashboard
                  </Button>
                </Link>
              ) : (
                <Link to="/login">
                  <Button
                    type="primary"
                    shape="round"
                    onClick={() => {
                      window.location.href = '/login';
                    }}
                  >
                    Login
                  </Button>
                </Link>
              )}
            </span>
          </QueueAnim>
          <div className={`${className}-image-wrapper`}>
            <BannerImage />
          </div>
        </div>
      </div>
    );
  }
}

export default Banner;
