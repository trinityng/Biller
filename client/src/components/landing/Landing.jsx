import React from 'react';
import DocumentTitle from 'react-document-title';
import Header from './Header';
import Banner from './Banner';
import Page1 from './Page1';
import Page2 from './Page3';
import Page3 from './Page2';
import Footer from './Footer';

import '../static/style.js';

class Landing extends React.PureComponent {
  render() {
    return (
      <div className="home-page">
        <Header key="header" />
        <Banner key="banner" />
        <Page1 key="page1" />
        <Page3 key="page3" />
        <Page2 key="page2" />  
        <Footer key="footer" />
        <DocumentTitle title="Biller" />
      </div>
    );
  }
}
export default Landing;
