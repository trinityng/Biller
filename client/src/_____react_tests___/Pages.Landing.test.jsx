import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Link } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import storageMock from './__mock__/MockLocalStore';

import * as consts from '../constants.js';

import Landing from '../components/landing/Landing';
import Header from '../components/landing/Header';
import Banner from '../components/landing/Banner';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('landing page / components', () => {
  let initialState = {};
  let store;

  beforeEach(() => {
    store = mockStore(initialState)
    window.localStorage = storageMock();
  });

  it('should render correctly', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <Landing />
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe("when user does not have a token", () => {
    it('should show register / login buttons', () => {
      const wrapper = shallow(
        <Banner />
      );

      const regButton = wrapper.find('.landing-register-button');
      const loginButton = wrapper.find('.landing-login-button');
      const dashboardButton = wrapper.find('.landing-dashboard-button');

      expect(regButton.length).toEqual(1);
      expect(loginButton.length).toEqual(1);
      expect(dashboardButton.length).toEqual(0);
    });
  });

  describe("when user has a token", () => {
    it('should show register / dashboard buttons', () => {
      localStorage.setItem(consts.USER_TOKEN, "mock");

      const wrapper = shallow(
        <Banner />
      );

      const regButton = wrapper.find('.landing-register-button');
      const loginButton = wrapper.find('.landing-login-button');
      const dashboardButton = wrapper.find('.landing-dashboard-button');

      expect(regButton.length).toEqual(1);
      expect(loginButton.length).toEqual(0);
      expect(dashboardButton.length).toEqual(1);
    });
  });
});