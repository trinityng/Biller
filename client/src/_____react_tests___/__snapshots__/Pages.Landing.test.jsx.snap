// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page / component should render correctly 1`] = `ReactWrapper {}`;
